release: release notes for v0.13.0 (#295)
confidential-containers/confidential-containers: releases/v0.13.0.md

> # Release Notes for v0.12.0

Author: whoops

Release Date: March 24th, 2025

This release is based on [3.15.0](https://github.com/kata-containers/kata-containers/releases/tag/3.13.0) of Kata Containers
and [v0.11.0](https://github.com/confidential-containers/enclave-cc/releases/tag/v0.11.0) of enclave-cc.

Please see the [quickstart guide](https://confidentialcontainers.org/docs/getting-started/) for details on how to try out Confidential
Containers.

Please refer to our [Acronyms](https://github.com/confidential-containers/documentation/wiki/Acronyms)
and [Glossary](https://github.com/confidential-containers/documentation/wiki/Glossary) pages for
definitions of the acronyms used in this document.

## What's new

* AMD SEV-SNP deployments are compatible with upstream host kernels.
* Trustee uses ECDH keys by default for KBS protocol.
* Images with many layers can be pulled without issues.

> * Images with whiteouts are pulled correctly.

Reviewer: Can a new section be added here, called `Bug Fix`? These two items can be added as
```
1. Fixed the issue where image pull failed due to exceeding the 4KB length limit on mount parameters caused by too many layers.
2. Fixed the issue of unexpected appearance of image whiteout files due to guest's tmpfs not supporting xattr.
```

Author: I moved these points to a bug fixes section but kept the wording fairly high-level.

Reviewer: Ok, I thought "without issues" is a bit absolute at first and bugfix could describe the problem more accurately and could help track related issues in the future. The High-level description is also acceptable. If it is High-level, we don’t need a new section and can just keep it as it is.

Reviewer: I approved the Pr. Fell free to choose the way you like

Author: Ok, I added the more specific stuff to the bug fixes section.

* The repository field of the resource URI is no longer optional.

> * TDX Attestation uses configfs instead of libtdx-attest.

Reviewer 2: ```suggestion
* TDX Attestation uses configfs-tsm based reports only and disables the libtdx-attest fallback.
```

* Trustee documentation added to project website.
* Trustee attestation policy improved for TDX.
* Blocking logs endpoint with policy can no longer cause deadlocks.

## Hardware Support

Attestation is supported and tested on three platforms: Intel TDX, AMD SEV-SNP, and IBM SE.

> Not all feature have been tested on every platform, but those based on attestation

Reviewer: ```suggestion
Not all features have been tested on every platform, but those based on attestation
```

are expected to work on the platforms above.

Make sure your host platform is compatible with the hypervisor and guest kernel

> provisioned by coco.

Reviewer: ```suggestion
provisioned by CoCo.
```

This release has been tested on the following stacks:

### AMD SEV-SNP

* Processor: AMD EPYC 7413
* Kernel: 6.12.0-snp-host-adc218676 (upstream 6.11+) 
* OS: Ubuntu 22.04.4 LTS
* k8s: v1.30.1 (Kubeadm)
* Kustomize: v4.5.4

### Intel TDX

* Kernel: [6.8.0-1004-intel](https://git.launchpad.net/~kobuk-team/ubuntu/+source/linux-intel/tree/?h=noble-main-next)
* OS: Ubuntu 24.04 LTS
* k8s: v1.30.2 (Kubeadm)
* Kustomize: v5.0.4-0.20230601165947-6ce0bf390ce3

### Secure Execution on IBM zSystems (s390x) running LinuxONE

* Hardware: IBM Z16 LPAR
* Kernel: 5.15.0-113-generic
* OS: Ubuntu 22.04.1 LTS
* k8s: v1.28.4 (Kubeadm)
* Kustomize: v5.3.0

## Limitations

The following are limitations and known issues with this release.

> * SEV(-ES) does not support attestation.

Reviewer 2: I forget what the conclusion was to notify users about the deprecation of these?

Author: We were a little vague about the plan but the reality is that the runners are off but most of the code is there. I updated the notes to say that it is deprecated and will be removed.
